python/cudf/cudf/core/dataframe.py

```diff
@@ -6482,11 +6519,15 @@ def select_dtypes(self, include=None, exclude=None):
             inclusion = set()
         # remove all exclude types
         inclusion = inclusion - exclude_subtypes
-
-        for k, col in self._data.items():
-            infered_type = cudf_dtype_from_pydata_dtype(col.dtype)
-            if infered_type in inclusion:
-                df._insert(len(df._data), k, col)
+        if inclusion:
+            for k, col in self._data.items():
+                infered_type = cudf_dtype_from_pydata_dtype(col.dtype)
+                if infered_type in inclusion:
+                    df._insert(len(df._data), k, col)
+        else:
```

Comment:
**suggestion (non-blocking, because it would be a bigger redo)**. Why is this function not implemented as:

1. determine columns to keep
2. `return self.loc[:, columns_to_keep]`

Then we could promote the handling of `self._data.rangeindex` and friends to `loc` (where it should presumably also be handled).